### ISSUE TITLE
More Versatile Waiting Methods

### ISSUE THREAD

**Issue by titusfortner** [watir/watir#458]
As discussed in #452 & #454, we should make it easier to call waits on elements.

Along the lines of:

``` ruby
element.wait_until(&:present?)
element.wait_while(&:visible?)
element.wait_until(&:enabled?)
element.wait_until(&:stale?)
element.wait_until { text == 'foo' }
element.when(&:enabled?).click
```

**Comment by titusfortner** [comment]
@p0deje - I'm not sure how to implement this so this in order to keep the error message without having to pass it in as a parameter. Did you have an idea?

**Comment by p0deje** [comment]
I don't think there is any way to keep error messages the way they are now. Let's just dump them.

Also, one example I've posted is invalid: 

``` ruby
element.wait_until { text == 'foo' }
```

should be

``` ruby
element.wait_until { |e| e.text == 'foo' }
```

**Comment by titusfortner** [comment]
I'm definitely not in favor of dumping the messages. If anything I'd like to see them made more specific and useful for debugging.

What about metaprogramming these? `define_method` the list of predicate methods the given element responds to; then we can use the `method` that is passed in as part of the message.

**Comment by p0deje** [comment]
Guess it's already done.
